rust/lance/src/io/exec/knn.rs

```diff
@@ -171,7 +171,7 @@ impl KNNFlatExec {
         if !is_vector {
             return Err(Error::IO {
                 message: format!(
-                    "KNNFlatExec node: query column {} is not a vector",
+                    "KNNFlatExec node: query column {} is not a vector (float32 fixed size array)",
                     query.column
```

Comment:
Maybe we could also name the data type we got?

```suggestion
                    "KNNFlatExec node: query column {} is not a vector. Expected float32 fixed-size array, got {}",
                    query.column, field.data_type()
```